dev/com.ibm.ws.org.apache.cxf.cxf.core.3.2/src/org/apache/cxf/common/logging/LogUtils.java

```diff
@@ -264,36 +264,64 @@ protected static Logger createLogger(final Class<?> cls,
                 try {
                     b = BundleUtils.getBundle(cls, bundleName);
                 } catch (MissingResourceException rex) {
-                    //ignore
                 }
             }
             if (b != null) {
                 b.getLocale();
             }
 
             if (loggerClass != null) {
-                try {
-                    Constructor<?> cns = loggerClass.getConstructor(String.class, String.class);
-                    if (name == null) {
+                // Default logger of Liberty is WsLogger. 
+                // CXFLogger is inherited from this WsLogger class to preserve same behavior
+                // except the one that we aimed to change. Which is to add class name to log records
+                if (loggerClass.equals(WsLogger.class)) {
+                    try {
+                        Constructor<?> cns = loggerClass.getConstructor(String.class, String.class, Class.class); //Liberty change: Class is added as parameter
+                        if (name == null) {
+                            try {
+                                return (CXFLogger) cns.newInstance(loggerName, bundleName, cls); //Liberty change: Class is added as parameter
+                            } catch (InvocationTargetException ite) {
+                                if (ite.getTargetException() instanceof MissingResourceException) {
+                                    return (CXFLogger) cns.newInstance(loggerName, null, cls); //Liberty change: Class is added as parameter
+                                }
+                                throw ite;
+                            }
+                        }
                         try {
-                            return (Logger) cns.newInstance(loggerName, bundleName);
+                            return (CXFLogger) cns.newInstance(loggerName, bundleName, cls); //Liberty change: Class is added as parameter
                         } catch (InvocationTargetException ite) {
                             if (ite.getTargetException() instanceof MissingResourceException) {
-                                return (Logger) cns.newInstance(loggerName, null);
+                                throw (MissingResourceException) ite.getTargetException();
                             }
                             throw ite;
                         }
+                    } catch (Exception e) {
+                        throw new RuntimeException(e);
```

Comment:
Since `CXFLogger` is Liberty only, should we fall back to the original code in the `else` statement rather than throw a RuntimeException? For example: something went wrong setting up our logging extension, so rather than fail out, we just fall back to the way we did it before. If their is still an exception caught in the original code, then its nothing to do with us.

Reply:
We could just add a logging statement saying we failed to create our enhanced logging instance, logging the exception message only, and then fall back to the original code.